meeshkan/sources/kafka.py

```diff
@@ -1,78 +1,62 @@
-from http_types.utils import HttpExchangeBuilder
-from typing import AsyncIterable
 import asyncio
-from .abstract import AbstractSource
-from typing import Tuple
+from dataclasses import dataclass
+from typing import AsyncIterable, Tuple
 
+import faust
+from http_types.types import HttpExchange
+from http_types.utils import HttpExchangeBuilder
+
+from .abstract import AbstractSource
 from ..meeshkan_types import HttpExchangeStream
-try:
-    import faust
-except ImportError:
-    faust = None
 
 
-class KafkaProcessorConfig:
+@dataclass(frozen=True)
+class KafkaSourceConfig:
     broker: str
     topic: str
 
-    def __init__(self, broker, topic):
-        self.broker = broker
-        self.topic = topic
-
-
-class KafkaProcessor:
-
-    def __init__(self, options: KafkaProcessorConfig):
-        self.options = options
-        self.app = faust.App('myapp', broker=options.broker,
-                             stream_wait_empty=False)
-        self.faust_topic = self.app.topic(
-            options.topic, key_type=str, value_type=str)
-
-        # self.stream = self.faust_topic.stream()
-
-        @self.app.agent(self.faust_topic, sink=[])
-        async def gen(recordings: AsyncIterable):
-            async for recording in recordings:
-                yield HttpExchangeBuilder.from_dict(recording)
 
-        self.agent = gen
-
-    @staticmethod
-    def run(app: faust.App, loop: asyncio.AbstractEventLoop, loglevel='info'):
-        worker = faust.Worker(app, loop=loop, loglevel=loglevel)
-
-        async def start_worker(worker: faust.Worker) -> None:
-            await worker.start()
-
-        try:
-            loop.run_until_complete(start_worker(worker))
-        finally:
-            worker.stop_and_shutdown()
+class KafkaSource(AbstractSource):
+    def __init__(self, config: KafkaSourceConfig):
+        self.app = faust.App('meeshkan-kafka-source',
+                              broker=config.broker,
+                              stream_wait_empty=False)
 
+        faust_topic = self.app.topic(config.topic, key_type=str, value_type=str)
 
-class KafkaSource(AbstractSource):
+        @self.app.agent(faust_topic)
+        async def recording_agent(stream):
+            """Dummy faust agent reading values from Kafka topic and passing values on.
 
-    def __init__(self, config: KafkaProcessorConfig):
+            Yields:
+                [dict] -- HttpExchange dictionary objects
+            """
+            async for rec in stream:
+                yield rec
```

Comment:
It would be nice if one could just `yield HttpExchange.from_dict(rec)` here but I have no idea how to access the "result stream". `agent.stream()` seems to returns the original Kafka input stream, so one could yield anything here...